extensions/spotify-beta/src/nowPlaying.tsx

```diff
@@ -0,0 +1,316 @@
+    const artistId = artists?.[0]?.id;
+    title = `${name} · ${artistName}`;
+
+    markdown = `# ${name}
+by ${artistName}
+
+![${name}](${albumImage}?raycast-width=250&raycast-height=250)
+`;
+
+    metadata = (
+      <Detail.Metadata>
+        <Detail.Metadata.Label title="Track" text={name} />
+        <Detail.Metadata.Label title="Artist" text={artistName} />
+        <Detail.Metadata.Label title="Album" text={albumName} />
+      </Detail.Metadata>
+    );
+
+    actions = (
+      <>
+        {trackAlreadyLiked && (
+          <Action
+            icon={Icon.HeartDisabled}
+            title="Dislike"
+            onAction={async () => {
+              await removeFromMySavedTracks({
+                trackIds: trackId ? [trackId] : [],
+              });
+              if (closeWindowOnAction) {
+                await showHUD(`Disliked ${name}`);
+                await popToRoot();
+                return;
+              }
+              await showToast({ title: `Disliked ${name}` });
+              await containsMySavedTracksRevalidate();
+            }}
+          />
+        )}
+        {!trackAlreadyLiked && (
+          <Action
+            icon={Icon.Heart}
+            title="Like"
+            onAction={async () => {
+              await addToMySavedTracks({
+                trackIds: trackId ? [trackId] : [],
+              });
+              if (closeWindowOnAction) {
+                await showHUD(`Liked ${name}`);
+                await popToRoot();
+                return;
+              }
+              await showToast({ title: `Liked ${name}` });
+              await containsMySavedTracksRevalidate();
+            }}
+          />
+        )}
+        <Action
+          icon={Icon.Forward}
+          title="Next"
+          shortcut={{ modifiers: ["cmd"], key: "arrowRight" }}
+          onAction={async () => {
+            await skipToNext();
+            if (closeWindowOnAction) {
+              await showHUD("Skipped to next");
+              await popToRoot();
+              return;
+            }
+            await showToast({ title: "Skipped to next" });
+            await currentPlayingRevalidate();
+          }}
+        />
+
+        <Action
+          icon={Icon.Rewind}
+          title="Previous"
+          shortcut={{ modifiers: ["cmd"], key: "arrowLeft" }}
+          onAction={async () => {
+            await skipToPrevious();
+            if (closeWindowOnAction) {
+              await showHUD("Skipped to previous");
+              await popToRoot();
+              return;
+            }
+            await showToast({ title: "Skipped to previous" });
+            await currentPlayingRevalidate();
+          }}
+        />
+        <Action
+          icon={Icon.Music}
+          title="Start Radio"
+          onAction={async () => {
+            await startRadio({
+              trackIds: trackId ? [trackId] : undefined,
+              artistIds: artistId ? [artistId] : undefined,
+            });
```

Comment:
If the `trackId` or `artistId` is `undefined`, what happens? If that's a possible case, the `Start Radio` action should probably not be shown.

Reply:
There will always be one of the ids defined, never none 👍